server/server.go

```diff
@@ -0,0 +1,393 @@
+	if err != nil {
+		return nil, err
+	}
+	// TODO: use session() NodeGetter or Peer NodeGetter methods directly?
+	node, err := s.peer.Get(ctx, cid)
+	if err != nil {
+		return nil, err
+	}
+
+	respNode, err := nodeToPbNode(node)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.GetNodeResponse{Node: respNode}, nil
+}
+
+func (s *ipfsLiteServer) GetNodes(req *pb.GetNodesRequest, srv pb.IpfsLite_GetNodesServer) error {
+	cids := make([]cid.Cid, len(req.GetCids()))
+	for i, cidString := range req.GetCids() {
+		cid, err := cid.Decode(cidString)
+		if err != nil {
+			return err
+		}
+		cids[i] = cid
+	}
+	// TODO: use session() NodeGetter or Peer NodeGetter methods directly?
+	ch := s.peer.GetMany(srv.Context(), cids)
+	for {
+		result, ok := <-ch
+		if ok == false {
+			break
+		} else {
+			resp := pb.GetNodesResponse{}
+			if result.Err != nil {
+				resp.Option = &pb.GetNodesResponse_Error{Error: result.Err.Error()}
+			} else {
+				node, err := nodeToPbNode(result.Node)
+				if err != nil {
+					resp.Option = &pb.GetNodesResponse_Error{Error: err.Error()}
+				} else {
+					resp.Option = &pb.GetNodesResponse_Node{Node: node}
+				}
+			}
+			srv.Send(&resp)
+		}
+	}
+	return nil
+}
+
+func (s *ipfsLiteServer) RemoveNode(ctx context.Context, req *pb.RemoveNodeRequest) (*pb.RemoveNodeResponse, error) {
+	cid, err := cid.Decode(req.GetCid())
+	if err != nil {
+		return nil, err
+	}
+	err = s.peer.Remove(ctx, cid)
+	if err != nil {
+		return nil, err
+	}
+	return &pb.RemoveNodeResponse{}, nil
+}
+
+func (s *ipfsLiteServer) RemoveNodes(ctx context.Context, req *pb.RemoveNodesRequest) (*pb.RemoveNodesResponse, error) {
+	cids := make([]cid.Cid, len(req.GetCids()))
+	for i, reqCid := range req.GetCids() {
+		cid, err := cid.Decode(reqCid)
+		if err != nil {
+			return nil, err
+		}
+		cids[i] = cid
+	}
+	err := s.peer.RemoveMany(ctx, cids)
+	if err != nil {
+		return nil, err
+	}
+	return &pb.RemoveNodesResponse{}, nil
+}
+
+func (s *ipfsLiteServer) ResolveLink(ctx context.Context, req *pb.ResolveLinkRequest) (*pb.ResolveLinkResponse, error) {
+	cid, err := cid.Decode(req.GetNodeCid())
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := s.peer.Get(ctx, cid)
+	if err != nil {
+		return nil, err
+	}
+
+	link, remainingPath, err := node.ResolveLink(req.GetPath())
+	if err != nil {
+		return nil, err
+	}
+
+	respLink := pb.Link{
+		Name: link.Name,
+		Size: int64(link.Size),
+		Cid:  link.Cid.String(),
+	}
+
+	resp := pb.ResolveLinkResponse{
+		Link:          &respLink,
+		RemainingPath: remainingPath,
+	}
+
+	return &resp, nil
+}
+
+func (s *ipfsLiteServer) Resolve(ctx context.Context, req *pb.ResolveRequest) (*pb.ResolveResponse, error) {
+	cid, err := cid.Decode(req.GetNodeCid())
+	if err != nil {
+		return nil, err
+	}
+
+	node, err := s.peer.Get(ctx, cid)
+	if err != nil {
+		return nil, err
+	}
+
+	// ToDo: Figure out how to use this _ interface{} data
+	_, remainingPath, err := node.Resolve(req.GetPath())
```

Comment:
Not sure how to deal with the data returned by resolve. It can be anything from a `Link` to a number or string or whatever data happens to be resolved. How do we represent that data and get it to the client?

Reply:
What about providing a tighter API like `ResolveLink`, `ResolveNumber`, `ResolveString`, `ResolveBytes`, etc.?

Reply:
(you'd error if the type check failed)

Reply:
Oh, I see there's already a `ResolveLink`

Reply:
I'm not sure if it's common to resolve anything but a Link? Is it?

Reply:
In theory, you normally just fetch the Node you're interested in and unmarshall its data into some known model data type

Reply:
What do you think about my theory @sanderpick? If it's the case that you'd normally not want/need to resolve anything but `Link`s, we could just not expose this `Resolve` method (in fact, it's not even exposed in the top level IPFS-lite api, I just chose to dig into and the `Node`'s api here).

Reply:
Yep, makes sense - come to think of it, I've never used Resolve at all... so, +1 for leaving it out.

Reply:
Sweeeeet